Bl 995 be i want to add edit a meeting ticket

JamesFincher opened the pull request:
## Description

NOTE---
This is only the PR for the FE update to this ticket. You can find the matching BE PR linked below.
[BE PR](https://github.com/BloomTech-Labs/underdog-devs-be-a/pull/167)
---------

With this ticket I decided to go big, and create not only the functionality requested from the ticket, but also build out what I feel is a solid scaffolding for a central meetings page. My decision to go this route was motivated by two questions. First, how might the user view and interact with the meetings. Second, how do we ensure that the available UI and permissions match what the type of user.

-I created three Modals. One to Create a Meeting, one to Update a meeting, and one to Delete a meeting. 

- @TylrPopcorn was able to get the styling and logic worked out for the Update and Delete modals. I copied his styles into the Create modal.

-I created three separate routes that correlate to the three user types who will need to access meetings. Admin, Mentor, and Mentee.

-The admin meeting dashboard can be accessed at '/admin/meetings'. This page provides three buttons so far that allow an admin to Create a new meeting, Update existing meetings, and Delete meetings. In the future, this can be hooked up to detect if a logged in user is Admin, and they will be directed to the proper area with open access to edit, update, and delete meetings.

-The mentor meeting dashboard can be accessed at '/mentor/meetings'. This page provides two buttons so far that allow a mentor to Create a new meeting, and Update existing meetings.  In the future, this can be hooked up to detect if a logged in user is a Mentor, and they will be directed to the proper area with limited access to create and edit their meetings. Furthermore, in the future, we would want to make sure the Mentor only has access to update their meetings, and can only create new meetings with their assigned Mentees.

-The mentee meeting dashboard can be accessed at '/mentee/meetings'. This page provides one button so far that allow a mentee to update any meetings with their Mentor. In the future, this can be hooked up to detect if a logged in user is a Mentee, and they will be directed to the proper area with limited access to edit  their meetings. Furthermore, in the future, we would want to make sure the Mentee only has access to update their meetings.

-I created new reusable component called DynamicDropdown. This reusable tool utilizes AntDesign's Select, and provides us a way to map data into a dropdown. A huge issue I had early on was realizing how strict the shape of the data being sent to the BE api was. As well, we don't currently have a way to abstract names from profile_id's. This dropdown provides a solid way to easily experiment with our meeting data on the front end, while also making sure the data being accessed and sent is valid.

- @joanterm helped me with a huge blocker and assisted me with the proper flow of props to get the DynamicDropdown component working properly.

- @TylrPopcorn helped me with code cleanup and polish.

-I struggled with how to structure the the three new dashboards. I ultimately decided to go ahead and build the admin dashboard first, and copy it into the Mentor and Mentee dashboards with minimal changes. My logic to this decision is.... I feel the project still has a bunch of features to be fleshed out, and that the overall structure will most likely change many more times in the development. I wanted to provide a general starting point for managing meetings, that can be abstracted and refactored in a way that makes sense as the project and it's requirements evolve. 


#### Video Link

[Loom Video](https://www.loom.com/share/542189ee74c04a13843d4b4ede21c2bb)

#### Jira Link

[Jira Ticket](https://bloomtechlabs.atlassian.net/browse/BL-995?atlOrigin=eyJpIjoiNjYxYmVhNDQzOWFmNGU3YTg2M2E4OTNmYTU3M2E4ODMiLCJwIjoiaiJ9)


## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have removed unnecessary comments/console logs from my code
- [x] My changes generate no new warnings
- [x] I have checked my code and corrected any misspellings
- [x] No duplicate code left within changed files
- [x] Size of pull request kept to a minimum
- [x] Pull request description clearly describes changes made & motivations for said changes

mina-mikhael reviewed (changes requested):
A lot of effort and time was spent to get the dashboard to look like what you showed in your Loom video. you all did an amazing job.

I would like to suggest some changes to resolve the conflict and to help wire your new dashboard with the BE.
- you need to make sure that you pull the changes from main to avoid merge conflicts. what got my attention was the axiosWithAuth hook in your branch is the older version. it's probably because you started working on the ticket, cloned, and synced with main and then these changes happened and were merged later.
- whenever you make an axios call to a protected endpoint on the BE you need to use the custom hook (useAxiosWithAuth0)  instead of regular axios, BE will require an access_token for all API calls and the hook will provide it.
- minor refactoring needs to be done in the dashboard components so they can work with the new axios hook.

I will try to point out some of the changes required in your code as much as I can in some comments for each file.

mina-mikhael commented on src/components/pages/AdminMeetingsDashBoard/AdminMeetingDash.js:
```diff
@@ -0,0 +1,159 @@
+  const [loading, setLoading] = useState(true);
+  const [allMentors, setAllMentors] = useState([]);
+  const [allMentees, setAllMentees] = useState([]);
+  const { axiosWithAuth } = useAxiosWithAuth0();
```
minor change required
const axiosWithAuth  = useAxiosWithAuth0();

JamesFincher replied:
I updated all instances new instances of axios to utilize the updated useAxiosWithAuth0 hook.

mina-mikhael commented on src/components/pages/AdminMeetingsDashBoard/AdminMeetingDash.js:
```diff
@@ -0,0 +1,159 @@
+  const { axiosWithAuth } = useAxiosWithAuth0();
+
+  const createNewMeeting = meeting => {
+    axiosWithAuth()
```
need to remove () from all axiosWithAuth API calls
axiosWithAuth.post('/meetings', meeting)

JamesFincher replied:
This has been done as part of the last commit updating all axios calls.

mina-mikhael commented on src/components/pages/AdminMeetingsDashBoard/CrudModals/CreateModal.js:
```diff
@@ -0,0 +1,212 @@
+import React, { useEffect, useState } from 'react';
+import 'antd/dist/antd.css';
+import { Button, Modal } from 'antd';
+import axios from 'axios';
```
should import useAxiosWithAuth0 instead of axios

JamesFincher replied:
This has been fixed. Thank you!

mina-mikhael commented on src/components/pages/AdminMeetingsDashBoard/CrudModals/CreateModal.js:
```diff
@@ -0,0 +1,212 @@
+      mentor_meeting_notes: formData.mentor_meeting_notes,
+      mentee_meeting_notes: formData.mentee_meeting_notes,
+    };
+    axios.post(`${process.env.REACT_APP_API_URI}meetings/`, meeting);
```
should use axiosWithAuth for all API Calls

JamesFincher replied:
This has been resolved in the last commit. Thank you!

mina-mikhael commented on src/components/pages/AdminMeetingsDashBoard/CrudModals/CreateModal.js:
```diff
@@ -0,0 +1,212 @@
+import DynamicDropdown from '../DynamicDropdown';
+
+const CreateModal = props => {
+  const { data, setMeetings, meetings } = props;
```
inside the function should invoke useAxiosWithAuth0 and save the instance to a variable, as following:

const axiosWithAuth = useAxiosWithAuth0()

JamesFincher replied:
This has been done in the last commit. Thank you!

mina-mikhael commented on src/components/pages/AdminMeetingsDashBoard/CrudModals/CreateModal.js:
```diff
@@ -0,0 +1,212 @@
+import React, { useEffect, useState } from 'react';
```
All comments on this file apply to the rest of modals and other files where you use axios instead of axiosWithAuth for API calls

JamesFincher replied:
Noted, I updated all the axios calls as requested in all instances. Thanks Mina!

JamesFincher commented:
> A lot of effort and time was spent to get the dashboard to look like what you showed in your Loom video. you all did an amazing job.
> 
> I would like to suggest some changes to resolve the conflict and to help wire your new dashboard with the BE.
> 
> * you need to make sure that you pull the changes from main to avoid merge conflicts. what got my attention was the axiosWithAuth hook in your branch is the older version. it's probably because you started working on the ticket, cloned, and synced with main and then these changes happened and were merged later.
> * whenever you make an axios call to a protected endpoint on the BE you need to use the custom hook (useAxiosWithAuth0)  instead of regular axios, BE will require an access_token for all API calls and the hook will provide it.
> * minor refactoring needs to be done in the dashboard components so they can work with the new axios hook.
> 
> I will try to point out some of the changes required in your code as much as I can in some comments for each file.

Thank you very much Mina for the feedback. I have updated the codebase to the latest version of main which should help the the merge conflict before. I have also updated all instances of axios to use the newer version of useAxiosWithAuth0.

ryan-hamblin commented on CODE_OF_CONDUCT.md:
```diff
@@ -14,22 +14,22 @@ appearance, race, religion, or sexual identity and orientation.
--   Gracefully accepting constructive criticism
--   Focusing on what is best for the community
--   Showing empathy towards other community members
+- Using welcoming and inclusive language
```
It looks like there were a bunch of `.md` files affected by a run of the prettier formatter tool. That's actually a good thing however there is a lot to review here now.

ryan-hamblin commented on src/components/pages/AdminMeetingsDashBoard/AdminMeetingDash.js:
```diff
@@ -0,0 +1,159 @@
+    getAllMentors();
+    getAllMentees();
+    getAllMeetings();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
why disable the line? instead I'd prefer we actually solve the problem the linter is warning us of.

ryan-hamblin commented on src/components/pages/AdminMeetingsDashBoard/CrudModals/CreateModal.js:
```diff
@@ -0,0 +1,215 @@
+        label: mentor.mentor_id,
+        name: mentor.mentor_id,
+      };
+
```
lets remove this empty line.

ryan-hamblin commented on src/components/pages/AdminMeetingsDashBoard/CrudModals/CreateModal.js:
```diff
@@ -0,0 +1,215 @@
+
+  const handleCancel = () => {
+    setIsModalOpen(false);
+  };
```
add empty line between 78 - 79 to follow format of the other functions in this file

ryan-hamblin commented on src/components/pages/AdminMeetingsDashBoard/CrudModals/CreateModal.js:
```diff
@@ -0,0 +1,215 @@
+
+  useEffect(() => {
+    mentorsMapped();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
please satisfy the linter in lieu of circumnavigating it.

ryan-hamblin commented on src/components/pages/AdminMeetingsDashBoard/CrudModals/CreateModal.js:
```diff
@@ -0,0 +1,215 @@
+
+  const handleAntChange = (value, option) => {
+    setFormData({ ...formData, [option.name]: value });
+  };
```
add line break

ryan-hamblin commented on src/components/pages/AdminMeetingsDashBoard/CrudModals/CreateModal.js:
```diff
@@ -0,0 +1,215 @@
+      ...formData,
+      [evt.target.id]: evt.target.value,
+    });
+  };
```
add line break

ryan-hamblin commented on src/components/pages/AdminMeetingsDashBoard/CrudModals/DeleteModal.js:
```diff
@@ -0,0 +1,89 @@
+
+  const showModal = () => {
+    setIsModalOpen(true);
+  };
```
add line break

ryan-hamblin commented on src/components/pages/AdminMeetingsDashBoard/CrudModals/DeleteModal.js:
```diff
@@ -0,0 +1,89 @@
+    setForm({
+      INPUT: '',
+    });
+
```
remove this empty line

ryan-hamblin reviewed (dismissed):
The work here seems super solid. I see a few instances where the formatter ran over some .md files so that's good. 

I also recognized some places where I'd like you to satisfy the linter instead of trying to disable it. The linter IS your friend... that you fight sometimes. 

Also there are some discrepancies in formatting concerning line breaks between methods/functions etc. Please clean up and you're groovy. Also fix the merge conflicts. :)

matt-arofin reviewed (approved):
A **lot** was introduced in this PR and I'm impressed that you put it all together coherently _and_ concisely. No glaring issues jump out at me, and it should work well with the changes you've made to the backend repo. Great work!

ramib5734 commented on src/components/pages/AdminMeetingsDashBoard/AdminMeetingDash.js:
```diff
@@ -97,7 +97,7 @@ const AdminMeetingDash = () => {
     getAllMentees();
     getAllMeetings();
-  }, []);
+  }, [axiosWithAuth]);
```
everything looks good so far!

jzkime commented on src/components/pages/AdminMeetingsDashBoard/CrudModals/CreateModal.js:
```diff
@@ -0,0 +1,216 @@
+      mentee_meeting_notes: formData.mentee_meeting_notes,
+    };
+    axiosWithAuth
+      .post(`${process.env.REACT_APP_API_URI}meetings/`, meeting)
```
I'm curious to know if the link in the env has a trailing `/` - probably does, but good to be aware as it could be an issue later.

jzkime commented on src/components/pages/AdminMeetingsDashBoard/CrudModals/UpdateModal.js:
```diff
@@ -0,0 +1,208 @@
+            />
+          </label>
+          <br />
+          {/** ------------ */}
```
Does this serve a purpose?

jzkime commented on src/components/pages/AdminMeetingsDashBoard/ViewAllMeetings/ViewAllMeetings.js:
```diff
@@ -0,0 +1,54 @@
+const ViewAllMeetings = () => {
+  const [meetingData, setMeetingData] = useState([]);
+
+  // commented out because it's slow and not working... yet
```
Good Todo note!

jzkime reviewed (approved):
Very impressive work, well done creating clean and concise code!
